File: Object oriented programming (II sem)/OOP in cpp/Fox.cpp
#include"Fox.h"

Fox::Fox(Position pos, World& world):
	Animal(FOX_NAME, FOX_STRENGTH, FOX_INITIATIVE, FOX_ICON, pos, world) {
}

void Fox::action() {
	Position newPos = pickRandomNeighbourField(STANDARD_RANGE);
	if (world.isEmpty(newPos)) {
		move(newPos);
	}
	else if (isStronger(world.getBoard()[newPos.x][newPos.y])) {
		collision(world.getBoard()[newPos.x][newPos.y]);
	}
	else return;
}

File: Object oriented programming (II sem)/OOP in java/Fox.java
import java.awt.*;

public class Fox extends Animal{
    public Fox(Position position, World world){
        super("Fox", 3, 7,new Color(255, 152, 0, 255), position, world);
    }

    @Override
    public void action(){
        Position desiredPosition = getDesiredPosition(1);
        if (world.isEmpty(desiredPosition)) {
            move(desiredPosition);
        } else if (isStrongerOrEqualThen(world.getOrganism(desiredPosition))) {
            collision(world.getOrganism(desiredPosition));
        }
    }
}
